dev/backcompat/defs.bzl

```
@@ -0,0 +1,133 @@
+"""Database schema backward compatibility definitions.
+
+Clones and patches Sourcegraph git repository under @sourcegraph_back_compat so we can run tests targets
+from that particular release against the new database schema.
+
+Flakes can be defined to skip known problematic tests which are either flaky or simply cannot run against
+the new schema in case of a breaking change. See //dev/backcompat:flakes.bzl for more details about
+how to define them.
+
+The final result is the definition of a @sourcegraph_back_compat target, whose test targets are exactly
+the same as back then, but with instead a new schema.
+
+Example: `bazel test @sourcegraph_back_compat//enterprise/internal/batches/...`.
+
+See https://sourcegraph.com/search?q=context:global+dev/backcompat/patches/back_compat_migrations.patch+repo:github.com/sourcegraph/sourcegraph+lang:Go&patternType=standard&sm=0&groupBy=repo
+for the command generating the mandatory patch file in CI for these tests to run.
+
+If the patch file were to be missing, a placeholder diff is in place to make it explicit (in the
+eventuality of someone running those locally).
+"""
+
+load("test_release_version.bzl", "MINIMUM_UPGRADEABLE_VERSION", "MINIMUM_UPGRADEABLE_VERSION_REF")
+load("flakes.bzl", "FLAKES")
+
+load("@bazel_gazelle//:deps.bzl", "go_repository")
+load("@bazel_tools//tools/build_defs/repo:git.bzl", "git_repository")
+
+# Shell snippet to disable a test on the fly. Needs to be formatted before being used.
+#
+# OSX ships BSD sed and the GNU sed that is traditionally available on Linux is named gsed instead.
+PATCH_GO_TEST = """_sed_binary="sed"
+if [ "$(uname)" == "Darwin" ]; then
+    _sed_binary="gsed"
+fi
+$_sed_binary -i "s/func {}/ func _{}/g" {}/*.go
```

Is this space intentional? Curious how this works :D

```suggestion
$_sed_binary -i "s/func {}/func _{}/g" {}/*.go
```

---

Typo on my end, I used to put `xxfunc` in there to triple check that it's messing up with the tests as intended and forgot about it. Thank you so much for catching this 🙏